core/client/fs/src/main/java/alluxio/client/block/policy/options/CreateOptions.java

```diff
@@ -21,21 +25,58 @@
  */
 @PublicApi
 public final class CreateOptions {
+  private AlluxioConfiguration mAlluxioConf;
   private String mLocationPolicyClassName;
   private int mDeterministicHashPolicyNumShards;
+  private TieredIdentity mTieredIdentity;
+  private long mBlockCapacityReserved;
+  private String mSpecificWorker;
 
   /**
+   * Creates a new instance of options for {@link alluxio.client.block.policy.BlockLocationPolicy}.
+   *
+   * Note that the setter for configuration is explicity excluded from this class. To override
+   * values in the options after calling {@code defaults()} programmers should use the respective
+   * setter.
+   *
+   * @param alluxioConf Alluxio configuration
    * @return the default {@link CreateOptions}
    */
-  public static CreateOptions defaults() {
-    return new CreateOptions();
+  public static CreateOptions defaults(AlluxioConfiguration alluxioConf) {
```

Comment:
Why are these called `CreateOptions`?

Reply:
Good question. This file was here long before. My assumption when working on this are they are the options needed to **create** a `BlockLocationPolicy` instance.

The Javadoc that was here prior says:

```
/**
 * Method options for creating a {@link alluxio.client.block.policy.BlockLocationPolicy}.
 */
```

Reply:
This seems like a good opportunity to change the name if you do not think it is appropriate.